x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -699,49 +699,24 @@ export function registerEncryptedSavedObjects(
       { key: 'ssl', dangerouslyExposeValue: true },
       { key: 'password', dangerouslyExposeValue: true },
     ]),
-    attributesToExcludeFromAAD: new Set([
-      'output_id',
-      'name',
-      'type',
-      'is_default',
-      'is_default_monitoring',
-      'hosts',
-      'ca_sha256',
-      'ca_trusted_fingerprint',
-      'config',
-      'config_yaml',
-      'is_internal',
-      'is_preconfigured',
-      'proxy_id',
-      'version',
-      'key',
-      'compression',
-      'compression_level',
-      'client_id',
-      'auth_type',
-      'connection_type',
-      'username',
-      'sasl',
-      'partition',
-      'random',
-      'round_robin',
-      'hash',
-      'topics',
-      'headers',
-      'timeout',
-      'broker_timeout',
-      'required_acks',
-      'preset',
-      'secrets',
+    attributesToIncludeInAAD: new Set([
+      'service_token',
+      'shipper',
+      'allow_edit',
+      'broker_ack_reliability',
+      'broker_buffer_size',
+      'channel_buffer_size',
     ]),
   });
   // Encrypted saved objects
   encryptedSavedObjects.registerType({
     type: MESSAGE_SIGNING_KEYS_SAVED_OBJECT_TYPE,
     attributesToEncrypt: new Set(['passphrase']),
+    attributesToIncludeInAAD: new Set(['private_key', 'public_key', 'passphrase_plain']),
```

Comment:
@dasansol92 Does these look good to you? 
I'm not seeing these properties in the [mappings](https://github.com/elastic/kibana/blob/61fe1307eb80551ba174c6ef25227cfa6d5c48a0/x-pack/plugins/fleet/server/saved_objects/index.ts#L665), are they missing intentionally?

Reply:
@joeypoon might have more specific answers for this. If I'm not wrong, that would mean those properties being part of the AAD which could not be desired as the `passphrase_plain` and the `token_plain` can eventually change.

Reply:
`passphrase_plain ` can change only if kibana goes from [no encryption key to encryption key](https://github.com/elastic/kibana/blob/cb211a8f304f9ecf3a91fafcfe229ced0fb9871a/x-pack/plugins/fleet/server/services/security/message_signing_service.ts#L286) but since this change is only from encrypted field to empty string field and remains unused after this change, we left it in AAD (encrypted -> unencrypted is not supported). in retrospect, it is probably safer to remove it from AAD but it being is AAD is current behavior.

`private_key` and `public_key ` remain unchanged and can stay in AAD.

> I'm not seeing these properties in the [mappings](https://github.com/elastic/kibana/blob/61fe1307eb80551ba174c6ef25227cfa6d5c48a0/x-pack/plugins/fleet/server/saved_objects/index.ts#L665), are they missing intentionally?

Yes, this is [intentional](https://github.com/elastic/kibana/pull/149785#discussion_r1104611130). We want them as binary so explicitly mapping the fields was unnecessary.

Reply:
> it is probably safer to remove it from AAD but it being is AAD is current behavior.

FYI, removing an existing attribute from AAD is not supported in serverless zero-downtime upgrades, unless the attribute is removed from the object type altogether.